### ISSUE TITLE
Local -W030 directive ignored

### ISSUE THREAD

**Issue by pigulla** [jshint/jshint#1852]
The following code causes the dreaded "Expected an assignment or function call and instead saw an expression." warning when it shouldn't. Encountered in IntelliJ (JSHint 2.5.4) and verified with the online version of JSHint:

``` javascript
/**
 * @class foo.Bar
 * @constructor
 */
function Bar() {
}

/* jshint -W030 */
/**
 * @type {string}
 */
Bar.prototype.baz;
/* jshint +W030 */

module.exports = Bar;
```

**Comment by shitcricket** [comment]
Still broken :(

**Comment by jugglinmike** [comment]
You can avoid the error for the time being with the `void` operator:

```js
/**
 * @class foo.Bar
 * @constructor
 */
function Bar() {
}

/**
 * @type {string}
 */
void Bar.prototype.baz;

module.exports = Bar;
```

Of course, this assumes your JSDoc parser still recognizes the instance method reference there. If so, this workaround might actually be preferable because it requires less ceremony.

**Comment by pigulla** [comment]
...or JSHint could actually fix this feature and make it work as advertised :-(

**Comment by jugglinmike** [comment]
Don't be sad! I said "for the time being" (and kept the issue open) because I recognize that this isn't ideal, but I thought it might help you for now.

**Comment by pigulla** [comment]
Sorry, didn't mean to be snappy :)

**Comment by rwaldron** [comment]
This is mostly tangential, but there's a smell in that code that I'd like to mention for the sake of "good code" discussion. 

```js
/**
 * @type {string}
 */
Bar.prototype.baz;
```

If this is a property whose value will be a string, it would make sense to (at least) define the initial value as an empty string. The JSDoc says `@type {string}`, but that value isn't a string by `typeof` or `instanceof`. 

```js
/**
 * @type {string}
 */
Bar.prototype.baz = "";
```

... Expresses the intention.

Anyway, still needs to be fixed in JSHint.

**Comment by pigulla** [comment]
This is true in this particular case. But oftentimes it's more like this (using Closure Compiler type annotations here):

```js
/**
 * @type {!Bar}
 */
Foo.prototype.bar;
```

Meaning that any `foo.bar` always is a `Bar` object (assigned in the constructor) and nothing else, in particular not `null` or `undefined`. Which means there is no sensible default value that can be used here.

**Comment by rwaldron** [comment]
Ah, indeed that's a great counter example, thanks for the follow up. 

> Which means there is no sensible default value that can be used here

I don't know if Closure Compiler allows it, but this what [`null` is for](http://es5.github.io/#x4.3.11).

**Comment by nicolo-ribaudo** [comment]
This is a duplicate of #2436
